Fix create namespace modal

MilanPospisil opened the pull request:
From Issue AAH-824:

- [x]  Resize modal to standard (shorter width)

- [x] Remove "please, provide a namespace" text under "name" field 

- [x]  Move popover question mark (input group) to form (next to field label) 

- [x]  Reword "name" field alert message from "Name can only contain [A-Za-z0-9_]" to "Name can only contain letters and numbers" 

- [x] Remove popover extra right padding 

- [x] Change "namespace owners" select prompt from "find a group" to "select a group", let selected groups appear in field as chips 

- [x] Add "Select namespace owner permissions" title above permissions section

Completed: 1, 2, 4, 6 (partialy), 7

I was not sure what 3, 5 and part of 6 ment to be, maybe it is already corrected.

himdel commented on src/components/permissions/permission-chip-selector.tsx:
```diff
@@ -30,10 +31,15 @@ export class PermissionChipSelector extends React.Component<IProps, IState> {
   }
 
   render() {
+    let selectPermissionCaption = this.props.selectPermissionCaption;
```
```suggestion
    let selectPermissionCaption = this.props.selectPermissionCaption || t`Select permissions`;
```

MilanPospisil replied:
Good idea.

himdel commented:
the test failures look relevant, probably want to update the tests :)

himdel commented:
* [x] Resize modal to standard (shorter width)

|before|after|
|-|-|
| ![20211116043958](https://user-images.githubusercontent.com/289743/141922079-4969b20c-1316-47f2-a321-f05a25e1fded.png) | ![20211116043847](https://user-images.githubusercontent.com/289743/141921961-3d40192a-a7c9-4223-80f9-ba68065e94f6.png) |

* [ ] Remove "please, provide a namespace" text under "name" field
* [ ] Move popover question mark (input group) to form (next to field label)
* [ ] Reword "name" field alert message from "Name can only contain [A-Za-z0-9_]" to "Name can only contain letters and numbers"
* [ ] Remove popover extra right padding
* [ ] Change "namespace owners" select prompt from "find a group" to "select a group", let selected groups appear in field as chips
* [ ] Add "Select namespace owner permissions" title above permissions section

himdel commented on src/components/namespace-modal/namespace-modal.tsx:
```diff
@@ -39,11 +39,8 @@ export class NamespaceModal extends React.Component<IProps, IState> {
     const name: string = this.state.newNamespaceName;
-
-    if (name == '') {
-      error['name'] = t`Please, provide the namespace name`;
```
I think this one needs to stay (and only the static helperText should be removed)

Otherwise typing something in and then removing it gives you `Name can only contain letters and numbers`

himdel commented:
Unless @ZitaNemeckova disagress, I'd just remove the `selectPermissionCaption` changes, as those don't seem to fit the description, and we can merge the rest, and address the unchecked point at the next UX meeting

himdel commented:
So, @MilanPospisil I talked with @sbuenafe-rh today, and we have couple of changes....

* the modal width should go to medium, not small (mostly because the permission select needs the space)
* "let selected groups appear in field as chips" is obsolete (from before you could set permissions for groups)
* "title above permissions section" - Susan will get back to us on this one, but the idea was ta add a form label before the groups

MilanPospisil commented:
Remove padding in helper text now looks like this. But this change is for all helpers, is it that you wanted or should I go with some props options in helper component?
![Screenshot_20211119_112655](https://user-images.githubusercontent.com/23277791/142608360-aa6f2216-b7ae-4bc7-8de8-846cb917689d.png)

himdel commented on test/cypress/integration/namespace_form.js:
```diff
@@ -32,20 +32,28 @@ describe('A namespace form', () => {
   });
 
   it('should give message before typing', () => {
```
the test name no longer reflects reality, looks like we should go with `should give message if input is empty`,
because the next test should be `should give message if input has incorrect characters` and those two are testing the same thing :)

himdel commented on src/components/namespace-modal/namespace-modal.tsx:
```diff
@@ -143,6 +142,7 @@ export class NamespaceModal extends React.Component<IProps, IState> {
             fieldId='groups'
             helperTextInvalid={this.state.errorMessages['groups']}
           >
+            {t`Select namespace owner permissions`}
```
this no longer belongs here, we're not  adding any title (https://github.com/ansible/ansible-hub-ui/pull/1219#issuecomment-970382928)

himdel commented:
> Remove padding in helper text now looks like this. But this change is for all helpers, is it that you wanted or should I go with some props options in helper component?

Yeah, for all helpers is good :+1: 

But it doesn't *look* that good yet, I'd still suggest making that a separate PR :)

The text is now too close to the close button vertically, I think we'd need to push the close button closer to the top right corner (and the same distance from each), and keep at least the same padding between the button and the text.

MilanPospisil commented:
So I removed the title and fixed the layout in helper. I hope it is all, Im starting to get lost in so much issues in one PR :-) Maybe I should divide this kind of problems in two PRs in future.

himdel reviewed (approved):
LGTM, I agee all the points are fixed now :+1: 

And we can talk about the proper way to solve all the other helper text paddings on the next UX meeting, the header you added works correctly here :)
